build-pyg-support.py

```diff
@@ -0,0 +1,283 @@
+#! /usr/bin/env python3
+
+import subprocess
+from threading import Timer
+import os
+import signal
+import tempfile
+import argparse
+import re
+import warnings
+
+PACKAGES = {
+    'torch': {
+        'tag': '1.10.1'
+    },
+    'cluster': {
+        'tag': '*',
+        'url': 'https://github.com/rusty1s/pytorch_cluster.git',
+        'conda_path': '/conda/pytorch-cluster/'
+    },
+    'scatter': {
+        'tag': '2.0.8',
+        'url': 'https://github.com/rusty1s/pytorch_scatter.git',
+        'conda_path': '/conda/pytorch-scatter/'
+    },
+    'sparse': {
+        'tag': '0.6.12',
+        'url': 'https://github.com/rusty1s/pytorch_sparse.git',
+        'conda_path': '/conda/pytorch-sparse/',
+        'metis': False
+    },
+    'spline-conv': {
+        'tag': '*',
+        'url': 'https://github.com/rusty1s/pytorch_spline_conv.git',
+        'conda_path': '/conda/pytorch-spline-conv/'
+    },
+    'pyg': {
+        'tag': '2.0.1',
+        'url': 'https://github.com/pyg-team/pytorch_geometric.git',
+        'conda_path': '/conda/pyg/'
+    }
+}
```

Comment:
You have a bunch of info here, but you still have special cases in the code below. Could you remove those special cases and put the information needed to separate cases here? Or if there is no general way to express the needs of the packages you could put the function in this structure directly.

Also please use `namedtuple` for the package data objects. That way you need fewer strings in the code and it's harder to make mistakes.

Reply:
Made a `dataclass `using `namedtuple`. See what you think.